.github/workflows/ci_cd.yml

```
@@ -32,15 +44,46 @@ jobs:
           java-version: '17'
           cache: 'gradle'
 
-      - name: Run build
+      - name: Check if main repo
+        id: is_main_repo
+        run: |
+          if [ $REPO_NAME == 'openremote/console-android' ]; then
+            echo "value=true" >> $GITHUB_OUTPUT
+          fi
+        env:
+          REPO_NAME: ${{ github.repository }}
+
+      - name: Run assemble
+        if: ${{ !steps.is_main_repo.outputs.value || github.event_name == 'pull_request' }}
+        run: |
+          gradle assemble
+
+      - name: Run assemble and publish
+        if: ${{ steps.is_main_repo.outputs.value && github.event_name != 'pull_request' }}
         run: |
-          cd GenericApp
-          gradle build
+          echo "$KEYSTORE_BASE64" | base64 -d > keystore
+          command="gradle assemble publishToSonatype closeSonatypeStagingRepository"
```

When we're confident everything works well, we can close and release the artifacts immediately using:

```suggestion
          command="gradle assemble publishToSonatype closeAndReleaseSonatypeStagingRepository"
```